### ISSUE TITLE
User imported from LDAP cannot be disabled in non-master realm

### ISSUE THREAD

**Issue by ljcvok** [keycloak/keycloak#35799]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ldap

### Describe the bug

In a non-master realm, it is impossible to disable a user when the user is federated using LDAP configured as UNSYNCED.
In the master realm, a user can be disabled even when the LDAP is configured in READ_ONLY mode.

There is some discrepancy between master and any other realms, despite having the same settings (both Realm settings and LDAP settings).

### Version

26.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

I would expect regardless of the realm that a user can be disabled if LDAP is configured as UNSYNCED.

### Actual behavior

In a newly created realm which copies the master realm in every configuration (realm settings and LDAP configuration), it is impossible to disable the user. The request goes through with no error in the log, but the user remains active.
I can disable users without error in the master realm and the change actually happens, the user is disabled.

### How to Reproduce?

- Configure the master realm to use LDAP (UNSYNCED) and import some users
- Create a new realm with the same settings as the master and mimic also LDAP user federation
- Try to disable user in master realm - it is ok
- Try to disable user in second realm - it is not ok, user remains active

**Comment by vramik** [comment]
Thank you @ljcvok for reporting the issue. 

I've tried this locally with current nightly using following steps:

- started server
- created new realm
- created LDAP user federation provider in the realm with edit mode `UNSYNCED`
- clicked on `Users`
- clicked on `bwilson` user 

the user comes from ldap:
![image](https://github.com/user-attachments/assets/f137fecb-a80c-4f3f-adee-6c80a45b4ce7)

- clicked `Disabled` and `Save`
- went back to `Users` and see that the user was daisabled
![image](https://github.com/user-attachments/assets/e0379547-d95d-4b62-a9d4-c125f0813433)


Therefore it seems that I'm not able to reproduce the issue locally. I'd need some steps how to do it to proceed.

~missing-info

**Comment by ljcvok** [comment]
@vramik You are right, I tested it once again and it works on a fresh installation, so I dug more where the problem could be.
It seems the problem is with READ_ONLY vs UNSYNCED mode. When you set up the LDAP first as READ_ONLY and change to UNSYNCED afterwards the user can't be disabled.
Already imported users during READ_ONLY mode probably stay in READ_ONLY mode forever regardless of the mode set in the LDAP provider.

**Comment by pedroigor** [comment]
The UX is not really the best when changing the sync mode after creating the provider. If you look at our documentation https://www.keycloak.org/docs/latest/server_admin/#storage-mode there is a note there recommending to decide the sync mode beforehand in order to avoid changing the sync mode later.

The main reason for that is that mappers are not updated when changing the sync mode at the provider level and that causes unexpected behaviors.

We won't improve this now but feel free to open an `enhancement` so that we can track it.
